### PR TITLE
[ASDisplayNode] Placeholders should always be recreated if returning to cleared nodes.

### PR DESCRIPTION
**Pull request by appleguy**
If previously-displayed contents is gone (e.g. clearContents), and is not finished displaying
by the time the node is onscreen, recreate the placeholder immediately.

### CODE REVIEW

**Comment by appleguy** [comment]
@maicki @garrettmoon (ASImageNode changes) @nguyenhuy it would be great to have a review from one or two of you. Until then, I am going to go ahead and land this, so that I can get increased test coverage over the weekend as I continue to do development on other branches that I would like to rebase with this change.

**Comment by appleguy** [comment]
cc @Adlai-Holler regarding the change for multiplex.  Note that placeholder fading out now works reliably as well.

File: AsyncDisplayKit/ASDisplayNode.mm [hunk 1]
```diff
@@ -1539,9 +1540,20 @@ - (void)__enterHierarchy
...
+    // If we don't have contents finished drawing by the time we are on screen, immediately add the placeholder (if it is enabled and we do have something to draw).
+    if (self.contents == nil) {
+      CALayer *layer = self.layer;
       [layer setNeedsDisplay];
+      
+      if ([self _shouldHavePlaceholderLayer]) {
+        [CATransaction begin];
+        [CATransaction setDisableActions:YES];
+        [self _setupPlaceholderLayerIfNeeded];
+        _placeholderLayer.opacity = 1.0;
```
Comment (maicki) [inline AsyncDisplayKit/ASDisplayNode.mm]: Are we sure that the placeholder is always opaque?

**Comment by maicki** [comment]
@appleguy Added one small comment but LGTM

**Comment by Adlai-Holler** [comment]
Love this! Nice @appleguy!

**Pull request merged**